trieste/models/keras/interface.py

```diff
@@ -31,24 +31,24 @@ class NeuralNetworkPredictor(ProbabilisticModel, tf.Module, ABC):
     toolbox.
     """
 
-    def __init__(self, optimizer: Optional[BatchOptimizer] = None):
+    def __init__(self, optimizer: Optional[Union[BatchOptimizer, KerasOptimizer]] = None):
         """
         :param optimizer: The optimizer wrapper containing the optimizer with which to train the
             model and arguments for the wrapper and the optimizer. The optimizer must
             be an instance of a :class:`~tf.optimizers.Optimizer`. Defaults to
             :class:`~tf.optimizers.Adam` optimizer with default parameters.
+        :raise ValueError: If the optimizer is not an instance of :class:`~tf.optimizers.Optimizer`.
         """
         super().__init__()
 
         if optimizer is None:
             optimizer = BatchOptimizer(tf.optimizers.Adam())
```

Comment:
Is this still the right default or should we use `KerasOptimizer`? (Should `NeuralNetworkPredictor` still support `BatchOptimizer`?)

Reply:
I have switched to KerasOptimizer